train/trainer.py

```diff
@@ -254,27 +260,58 @@ def reconfigure_optimizers(self):
         optimizers, _ = self.configure_optimizers(
             existing_scheduler=self.lr_schedulers())
         self.optimizers()._optimizer = optimizers[0]
-
+    
+    def get_optimizer_params(self, weight_decay):
```

Comment:
make this function work correctly even when weight_decay = 0, by adding an if at the beginning eg
```
if weight_decay <= 0:
    return self.parameters()
````